Extensions/Multiplayer/multiplayertools.ts

```diff
@@ -0,0 +1,1054 @@
+              }
+              handleLobbyUpdatedEvent(runtimeScene, lobby, positionInLobby);
+              break;
+            }
+            case 'gameCountdownStarted': {
+              handleGameCountdownStartedEvent(runtimeScene);
+              break;
+            }
+            case 'gameStarted': {
+              handleGameStartedEvent(runtimeScene);
+              break;
+            }
+            case 'gameEnded': {
+              handleGameEndedEvent();
+              break;
+            }
+            case 'peerId': {
+              const messageData = messageContent.data;
+              if (!messageData) {
+                logger.error('No message received');
+                return;
+              }
+              const peerId = messageData.peerId;
+              if (!peerId) {
+                logger.error('Malformed message received');
+                return;
+              }
+
+              handlePeerIdEvent(peerId);
+              break;
+            }
+          }
+        }
+      };
+      _websocket.onclose = () => {
+        logger.info('Disconnected from the lobby.');
+
+        _connectionId = null;
+        playerNumber = null;
+        _lobbyId = null;
+        _lobby = null;
+        _lobbyOnGameStart = null;
+        _websocket = null;
+        if (_heartbeatInterval) {
+          clearInterval(_heartbeatInterval);
+        }
+
+        const lobbiesIframe = multiplayerComponents.getLobbiesIframe(
+          runtimeScene
+        );
+
+        if (!lobbiesIframe || !lobbiesIframe.contentWindow) {
+          return;
+        }
+
+        // Tell the Lobbies iframe that the lobby has been left.
+        lobbiesIframe.contentWindow.postMessage(
+          {
+            id: 'lobbyLeft',
+          },
+          '*' // We could restrict to GDevelop games platform but it's not necessary as the message is not sensitive, and it allows easy debugging.
+        );
+      };
+    };
+
+    const handleConnectionIdReceived = function ({
+      runtimeScene,
+      connectionId,
+      positionInLobby,
+      lobbyId,
+      playerId,
+      playerToken,
+      validIceServers,
+    }: {
+      runtimeScene: gdjs.RuntimeScene;
+      connectionId: string;
+      positionInLobby: number;
+      lobbyId: string;
+      playerId: string;
+      playerToken: string;
+      validIceServers: {
+        urls: string;
+        username?: string;
+        credential?: string;
+      }[];
+    }) {
+      // When the connectionId is received, initialise PeerJS so players can connect to each others afterwards.
+      if (validIceServers.length) {
+        console.info('Using custom servers:', validIceServers);
+        for (const server of validIceServers) {
+          gdjs.evtTools.p2p.useCustomICECandidate(
+            server.urls,
+            server.username,
+            server.credential
+          );
+        }
+      }
+      gdjs.evtTools.p2p.useDefaultBrokerServer();
+      // gdjs.evtTools.p2p.useCustomBrokerServer(
```

Comment:
Probably a good idea to have it sent from the backend (either not specified, in case useDefaultBrokerServer, or if passed then useCustomBrokerServer, with all parameters that can be set by the backend)

Reply:
done